src/main/java/uk/gov/justice/hmpps/prison/repository/jpa/model/OffenderAssessment.java

```diff
@@ -0,0 +1,185 @@
+    private LocalDate assessmentDate;
+
+    @Column(name = "ASSESS_COMMENT_TEXT")
+    private String assessmentComment;
+
+    @ManyToOne(fetch = FetchType.LAZY)
+    @JoinColumn(name = "ASSESSMENT_CREATE_LOCATION")
+    private AgencyLocation assessmentCreateLocation;
+
+    @Column(name = "ASSESS_STATUS")
+    private String assessStatus;
+
+    @ManyToOne(fetch = FetchType.LAZY)
+    @NotFound(action = IGNORE)
+    @JoinColumnsOrFormulas(value = {
+        @JoinColumnOrFormula(formula = @JoinFormula(value = "'" + AssessmentCommittee.ASSESS_COMMITTEE + "'", referencedColumnName = "domain")),
+        @JoinColumnOrFormula(column = @JoinColumn(name = "ASSESS_COMMITTE_CODE", referencedColumnName = "code"))
+    })
+    private AssessmentCommittee assessCommittee;
+
+    @ManyToOne(fetch = FetchType.LAZY)
+    @NotFound(action = IGNORE)
+    @JoinColumnsOrFormulas(value = {
+        @JoinColumnOrFormula(formula = @JoinFormula(value = "'" + AssessmentOverrideReason.OVERRIDE_REASON + "'", referencedColumnName = "domain")),
+        @JoinColumnOrFormula(column = @JoinColumn(name = "OVERRIDE_REASON", referencedColumnName = "code"))
+    })
+    private AssessmentOverrideReason overrideReason;
+
+    @ManyToOne(fetch = FetchType.LAZY)
+    @NotFound(action = IGNORE)
+    @JoinColumnsOrFormulas(value = {
+        @JoinColumnOrFormula(formula = @JoinFormula(value = "'" + AssessmentCommittee.ASSESS_COMMITTEE + "'", referencedColumnName = "domain")),
+        @JoinColumnOrFormula(column = @JoinColumn(name = "REVIEW_COMMITTE_CODE", referencedColumnName = "code"))
+    })
+    private AssessmentCommittee reviewCommittee;
+
+    @Column(name = "COMMITTE_COMMENT_TEXT")
+    private String reviewCommitteeComment;
+
+    @Column(name = "NEXT_REVIEW_DATE")
+    private LocalDate nextReviewDate;
+
+    @Column(name = "EVALUATION_DATE")
+    private LocalDate evaluationDate;
+
+    @OneToOne(fetch = FetchType.LAZY)
+    @JoinColumn(name="CREATION_USER")
+    private StaffUserAccount creationUser;
+
+    @OneToOne(fetch = FetchType.LAZY)
+    @JoinColumn(name="ASSESSMENT_TYPE_ID")
+    private AssessmentEntry assessmentType;
+
+    @OneToMany(fetch = FetchType.LAZY)
+    @JoinColumns({
+        @JoinColumn(name="OFFENDER_BOOK_ID", referencedColumnName="OFFENDER_BOOK_ID"),
+        @JoinColumn(name="ASSESSMENT_SEQ", referencedColumnName="ASSESSMENT_SEQ")
+    })
+    private List<OffenderAssessmentItem> assessmentItems;
+
+    @EqualsAndHashCode
+    @NoArgsConstructor
+    @AllArgsConstructor
+    @ToString
+    public static class Pk implements Serializable {
+        private Long bookingId;
+        private Integer assessmentSeq;
+    }
+
+    /**
+     * Method to generate a summary of the classification outcome.
+     */
+    @NonNull
+    public ClassificationSummary getClassificationSummary() {
+        if (reviewedClassification != null && !reviewedClassification.isPending()) {
+            AssessmentClassification previousClassification = null;
```

Comment:
Probably just personal preference but I wonder if you extracted the logic for getting a previous assessment it'll look for functional.

Reply:
ok